db/config.cc

```diff
@@ -115,80 +115,154 @@ config_from_string(std::string_view value) {
 }
 
 template <>
-const config_type config_type_for<bool> = config_type("bool", value_to_json<bool>);
+const config_type& config_type_for<bool>() {
```

Comment:
Is this not a series of instantiations where a macro might be warranted? If for nothing else so to make it easier to change if ever needed.... Seems a bit repetitive...

Reply:
I dislike boilerplate, but I dislike the preprocessor even more.